### ISSUE TITLE
Isomorphic export with `solidjs/web`

### ISSUE THREAD

**Issue by TomokiMiyauci** [solidjs/solid#2087]
### Describe the bug

There are dev.js, server.js and web.js as dist of `solid-js/web`.
These do not export the same members.

This affects the viability of projects that have this as a peer dependency.

### Your Example Website or App

No

### Steps to Reproduce the Bug or Issue

1. install
```bash
pnpm add solid-js @solidjs/router
```

2. run script

```main.js
import * as module from "@solidjs/router";

console.log(module);
```

### Expected behavior

I expected it to run successfully, but the following error occurs:

```
import { isServer, getRequestEvent, createComponent as createComponent$1, memo, delegateEvents, spread, mergeProps as mergeProps$1, template } from 'solid-js/web';
                                                                          ^^^^
SyntaxError: The requested module 'solid-js/web' does not provide an export named 'memo'
```

### Platform

- OS: macOS

### Additional context

solid-js: 1.8.15
@solidjs/router: 0.12.4

---
Looking at the following, I surmised that the project was aimed at isomorphic exports.

https://github.com/solidjs/solid/blob/3212f74db68a526754ce107f1d3e1da0809c4678/packages/solid/web/server/index.ts#L21-L26

**Comment by ryansolid** [comment]
Hmm... yeah ones that I figured people might use directly I added there. The `memo` is getting pulled in because it is from the compiled output of the router's JSX. Specifically the client build so I think there is a mismatch here. Generally we ship source with our libraries so that particularly export would never really be hit.

It does bring a question whether empty server exports should just be added to dom-expressions for all the methods it could generate during client build. Mind you it is very unlikely that any probably constructed project would hit this. Or rather it would like you did above you'd find it fail as soon as you tried to run it anyway (as the router in that build would be trying to access the DOM on the server where it isn't present).

The recommendation here is to setup your project with export conditions set to "solid" in addition to whatever environment ones you have.

**Comment by TomokiMiyauci** [comment]
@ryansolid One option is to not export the same.
In that case, I recommend removing the export noted above.

Also, there seems to be an error in the type mapping.

```ts
import * as module from "solid-js/web";
console.log(module.memo);
```

Intellisense infers the type as if ``memo`` existed. (without changing the `conditions`).

**Comment by ryansolid** [comment]
Yeah I'm not even sure how to get those types to work given the code runs on both sides. An IDE could never choose right. Other than a small handful of exports most things from `solid-js/web` should probably not be imported directly. The ones that could I added to the list. The rest I didn't bother. Like `render` definitely could be called directly so it needs to be exported from both.

That being said the only reason the original issues happens is because the client build of the router is getting pulled into the server, which would never work anyway. So I'm not sure there is an option other than stubbing everything, but even then it doesn't give any indicator of what you can and can't use. So I'm inclined to leave things as is.

Maybe `solid-js/web` could have like an internal sub path to indicate things that shouldn't be imported directly and that the compiler uses. Something to take into consideration for the future but I think there is no action now.
